### PR TITLE
workflow engine, assign role and update status 

### PR DESCRIPTION
**Pull request by nikhilchandrabackcountry**

### CODE REVIEW

File: BC-CMS.postman_collection.json [hunk 1]
```diff
@@ -15,7 +15,7 @@
 						"header": [],
```
Comment (RajKumar-BackCountry) [inline BC-CMS.postman_collection.json]: Remove this file

File: README.md [hunk 2]
```diff
@@ -2,25 +2,25 @@
```
Comment (RajKumar-BackCountry) [inline README.md:2]: Try adding some points about the apis as well.

File: prisma/schema.prisma [hunk 3]
```diff
@@ -68,11 +68,11 @@ model workflow {
   editor              String?
```
Comment (RajKumar-BackCountry) [inline prisma/schema.prisma:68]: Did you connect with Seby about adding both user's displayName as well as user's unique id ?

File: src/modules/workflows/workflowEngine.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
+/* eslint-disable indent */
+const { Status, WorkflowKeysEnum } = require('./enums');
+
+module.exports = (workflow, { writer, editor }) => {
+  const currentStatus = workflow.status;
+  const changeLog = {};
+
+  switch (currentStatus) {
+    case Status.WAITING_FOR_WRITER:
```
Comment (RajKumar-BackCountry) [inline src/modules/workflows/workflowEngine.js:9]: this is the very first status that will be assigned to a workflow (during creation).
While creating, we won't have the writer's information.
So, this error will be thrown always.
is that the expectation ?

File: src/modules/workflows/workflowEngine.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+    case Status.ASSIGNED_TO_WRITER:
+      if (writer) {
+        changeLog[WorkflowKeysEnum.writer] = writer;
+      } else {
+        changeLog[WorkflowKeysEnum.status] = Status.WRITING_IN_PROGRESS;
+      }
+      break;
+
+    case Status.WRITING_IN_PROGRESS:
```
Comment (nikhilchandrabackcountry) [inline src/modules/workflows/workflowEngine.js:25]: how to handle save for later?

asignedToWriter ---> saveforlater -> next
writinginpreogress-- > saveforlater -> auditlog

asignedToWriter ---> saveforlater -> next
writinginpreogress-- > saveforlater (wip) -> next

File: src/modules/workflows/workflowEngine.js (hunk 4 above)
Comment (nikhilchandrabackcountry) [reply src/modules/workflows/workflowEngine.js:9]: as discussed, this is the current status, to move to the next state we need writer info

File: src/modules/workflows/workflowEngine.js [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+      if (writer) {
+        changeLog[WorkflowKeysEnum.writer] = writer;
+        changeLog[WorkflowKeysEnum.status] = Status.ASSIGNED_TO_WRITER;
+      } else if (saveForLater) {
```
Comment (nikhilchandrabackcountry) [inline src/modules/workflows/workflowEngine.js:29]: Save for later logic comes here

File: src/modules/workflows/workflowEngine.js [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+module.exports = (workflow, { writer, editor }, saveForLater) => {
+  const currentStatus = workflow.status;
+  const changeLog = {};
+
+  switch (currentStatus) {
+    case Status.WAITING_FOR_WRITER:
+      if (!writer) {
+        throw new Error('A writer must be provided to proceed to the next step.');
+      }
+      changeLog[WorkflowKeysEnum.writer] = writer;
```
Comment (nikhilchandrabackcountry) [inline src/modules/workflows/workflowEngine.js:13]: can an editor be assigned here?

**Pull request closed without merging**